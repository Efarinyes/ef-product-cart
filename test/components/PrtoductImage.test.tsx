import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';


describe('ProductImage', () => {
    test('Ha de mostrar el component de manera correcta i amn la imatge personalitzada', () => {
        const wrapper = renderer.create(
            <ProductImage img="https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png" />
        )
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('Ha de mostrar el prodicte amb imatgecorrectament', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        )
        expect(wrapper.toJSON()).toMatchSnapshot();
    })
})