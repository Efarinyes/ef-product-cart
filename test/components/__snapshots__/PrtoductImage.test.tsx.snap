// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Ha de mostrar el component de manera correcta i amn la imatge personalitzada 1`] = `
<img
  alt="Product Foto"
  className="productImg undefined"
  src="https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png"
/>
`;

exports[`ProductImage Ha de mostrar el prodicte amb imatgecorrectament 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product Foto"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
